rules/no-unnecessary-polyfills.js

```diff
@@ -0,0 +1,250 @@
+'use strict';
+const semver = require('semver');
+const readPkgUp = require('read-pkg-up');
+const coreJsCompat = require('core-js-compat');
+const {camelCase, upperFirst} = require('lodash');
+const {STATIC_REQUIRE_SELECTOR} = require('./selectors/index.js');
+
+const {data: compatData, entries: coreJsEntries} = coreJsCompat;
+
+const MESSAGE_ID_POLYFILL = 'unnecessaryPolyfill';
+const MESSAGE_ID_CORE_JS = 'unnecessaryCoreJsModule';
+const messages = {
+	[MESSAGE_ID_POLYFILL]: 'Use the built-in `{{featureName}}`.',
+	[MESSAGE_ID_CORE_JS]: 'All polyfilled features imported from `{{coreJsModule}}` are disponible as built-ins. Use the built-ins instead.',
+};
+
+function getTargetsFromPackage(cwd, useBrowserlistFieldByDefault) {
+	const result = readPkgUp.sync({cwd});
+	if (!result || !result.pkg) {
+		return;
+	}
+
+	const {browserlist} = result.pkg;
+	const nodeEngine = result.pkg.engines && result.pkg.engines.node && new SemverNodeVersion(result.pkg.engines.node);
```

Comment:
Not every package.json have `engine` field.

Reply:
Indeed, that is in case the `targets` field is not defined in the config. If the `engines` field is also undefined, the plugin is not executed.